inc/mapgeolocation.class.php

```diff
@@ -83,7 +83,9 @@ public function showMap() {
       }
 
       $(function(){
-         map_elt = initMap($('#setlocation_container'), 'setlocation', '200px');
+         if(typeof map_elt === 'undefined') {
```

Comment:
If `map_elt` is already initialized, then I gess all event handlers are also initialized. I have not tested it, but I guess L90 to L123 should also be included in this `if` block.